### PR TITLE
Make latitude, longitude, altitude and time settable via the web UI

### PR DESCRIPTION
**Pull request by mrosseel**
This helps in case of either forgetting your GPS dongle, or having a malfunctioning internal GPS.

Possible future work:
- it might be nice to have preset locations like HOME, Observatory, ... similar to how the different wifi networks are configured.

### CODE REVIEW

File: python/views/header.tpl [hunk 1]
```diff
@@ -25,6 +26,7 @@
         <li><a href="/">Home</a></li>
         <li><a href="/remote">Remote</a></li>
         <li><a href="/network">Network Setup</a></li>
+        <li><a href="/gps">GPS Setup</a></li>
```
Comment (brickbots) [inline python/views/header.tpl:29]: I'd rather have this on the 'Tools' page as it should (ideally) be rarely used.  I know it's come up a lot in the discord (people not using GPS, issues with locks), but I hope these are not a representative sample of users. If people are using this often I have failed in some significant way as it's absolutely NOT the intended (or a good) user experience.

**Review by brickbots — COMMENTED** [review]
Thank you so much for working to add this!  I'm excited to actually try this and see how it works in context.

File: python/PiFinder/server.py [hunk 2]
```diff
@@ -356,24 +398,24 @@ def serve_pil_image():
 
         @app.route("/gps-lock")
```
[the remark this answers is not in the document]
Comment (mrosseel) [reply python/PiFinder/server.py:399]: they used to be called with the old web interface, there were dedicated buttons for testing. Now it's probably no longer needed, I'll see if I can remove it.

File: python/views/header.tpl (hunk 1 above)
Comment (mrosseel) [reply python/views/header.tpl:29]: then I'd need to integrate it with the Tools page, and it's a lot of code, tools might become a garbage bin then. Maybe not adding it to the menu is enough? On save it returns to home

Comment (brickbots) [reply python/views/header.tpl]: Yes... tools is probably not the right place.  How about not on the menu, but accessible from the main page via the edit icon?

Comment (mrosseel) [reply python/views/header.tpl]: that's what I did in the last commit

File: python/views/index.tpl [hunk 3]
```diff
@@ -18,6 +18,7 @@
       <i class="material-icons medium">{{gps_icon}}</i>
     </td>
     <td class="grey-text text-lighten-1">{{gps_text}}<br>lat: {{lat_text}} / lon: {{lon_text}}</td>
+    <td><a href="/gps" class="grey-text"><i class="material-icons">edit</i></a></td>
```
Comment (brickbots) [inline python/views/index.tpl:21]: Sorry @mrosseel I missed this edit to the index page!  Merging now!

**Pull request merged**